### PR TITLE
Frontend dockerized for development

### PR DESCRIPTION
**Pull request by erwagasore**
I saw that both `alerting` and `api-flask` are dockerized and being accustomed to using Docker in development. I opted on dockerizing the frontend for development and maybe in the future we even set up for `production`.

The directory arrangement for this `mono-repo` can be re-arranged for better separation, check the image below.

directory_structure

### CODE REVIEW

**Comment by erwagasore** [comment]
@ericboucher please review

File: web/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM node:12-alpine
+
+WORKDIR /home/app
+COPY . .
+
+RUN addgroup -S appgroup
+RUN adduser -S -D -h /home/app app appgroup
+RUN chown -R app:appgroup /home/app
```
Comment (ericboucher) [inline web/Dockerfile:8]: Do we need these?

File: web/.dockerignore [hunk 2]
```diff
@@ -0,0 +1,4 @@
+**/node_modules
+
+alerting
```
Comment (ericboucher) [inline web/.dockerignore:3]: these are not needed anymore with this folder structure

File: web/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+version: '3.7'
```
Comment (ericboucher) [inline web/docker-compose.yml:1]: How about having just one docker-compose file in the main folder?

File: web/Dockerfile (hunk 1 above)
Comment (ericboucher) [reply web/Dockerfile:8]: Open quetsion, do we even need a Dockerfile?

How about having a named volumes for node modules and running yarn on launch?

Your approach works but it means that we will have to re-build the docker whenever we change a library

**Review by ericboucher — COMMENTED** [review]
It's a good start! I left a few comments inline, plus:

Don't move the README for now or split it into a main README and a more specific web README

File: web/.dockerignore (hunk 2 above)
Comment (erwagasore) [reply web/.dockerignore:3]: True

File: web/docker-compose.yml (hunk 3 above)
Comment (erwagasore) [reply web/docker-compose.yml:1]: Do you mean for all services "web", "alerting", and "api-flask"?

File: web/Dockerfile (hunk 1 above)
Comment (erwagasore) [reply web/Dockerfile:8]: For development, we don't need them.

File: web/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+      context: .
+      dockerfile: Dockerfile
+    image: prism_web:latest
+    container_name: prism_web
+    env_file:
+        - .env
+    volumes:
+      - '.:/home/app'
+    ports:
+      - ${PORT}:${PORT}
```
[the remark this answers is not in the document]
Comment (erwagasore) [reply web/docker-compose.yml:19]: Got it.

**Comment by ericboucher** [comment]
Can I propose this structure:
- website
- api
- alerting

**Comment by wadhwamatic** [comment]
As discussed today @erwagasore , let's separate the directory restructure from dockerization.

**Comment by ericboucher** [comment]
Let's close this PR, it has diverged too much anyways and it will be easier to re-open a new one.

**Pull request closed without merging**